add custom replication port

stromp opened the pull request:
Allow to run the cluster on a non-standard replication port

stromp committed:
Add custom replication port.

- Add attribute to configure replication port
- Configure gcomm address by Find port via search
- Add SELinux policies

shoekstra commented on recipes/galera.rb:
```diff
@@ -110,7 +110,7 @@
     gcomm += ',' unless first
-    gcomm += lnode['fqdn']
+    gcomm += "#{lnode['fqdn']}:#{node['mariadb']['galera']['cluster_replication_port']}"
     first = false
```
`node['mariadb']['galera']['cluster_replication_port']` should be `lnode['mariadb']['galera']['cluster_replication_port']` yes?  This way it uses the attribute in the node data found by the search and not what is defined locally.

The way it is now, if node1 has `['mariadb']['galera']['cluster_replication_port'] = 5678` and node2 has `['mariadb']['galera']['cluster_replication_port'] = 6789`, node1 will add `node2:5678` to the array which wouldn't work.

shoekstra commented on attributes/default.rb:
```diff
@@ -99,6 +99,7 @@
 default['mariadb']['galera']['cluster_name'] = 'galera_cluster'
 default['mariadb']['galera']['cluster_search_query'] = ''
+default['mariadb']['galera']['cluster_replication_port'] = '4567'
 # All Galera nodes should get the same server_id
```
This attribute isn't used to configure what port to listen on for replication traffic - is something missing from this commit?

dimsh99 committed:
Fix for issue #128

dimsh99 committed:
Changes in specs

shoekstra committed:
Fix cookstyle warnings

shoekstra committed:
Make use of Chef::Util::Selinux to test selinux

shoekstra committed:
More custom replication port stuff

- Tweak Galera recipe code
- Fix/add some ChefSpec
- Fix some Cookstyle warnings

shoekstra commented:
Rebased and fixed some things, let's see how the tests go now..
